gateway/gateway.go

```diff
@@ -308,11 +307,13 @@ func (gateway *HandleT) dbWriterWorkerProcess(process int) {
 		for _, userWorkerBatchRequest := range breq.batchUserWorkerBatchRequest {
 			jobList = append(jobList, userWorkerBatchRequest.jobList...)
 		}
+
+		ctx, cancel := context.WithTimeout(context.Background(), WriteTimeout)
```

Comment:
@lvrach some more advanced timeout calculation could happen, taking into account the time that has passed since the first http request of the batch was received, [example here](https://github.com/rudderlabs/rudder-server/commit/6253de56a00930c640a96b4f364548061313752b), but I don't think it is really worth the extra complexity.